native/src/seal/decryptor.cpp

```diff
@@ -438,6 +438,43 @@ namespace seal
         StrideIter<const uint64_t *> wide_noise_poly((*noise_poly).ptr(), coeff_modulus_size);
         poly_infty_norm_coeffmod(wide_noise_poly, coeff_count, context_data.total_coeff_modulus(), norm.get(), pool_);
 
+        return norm;
+    }
+
+    double Decryptor::invariant_noise(const Ciphertext &encrypted) {
+        double invariant_noise = 0.0;
+
+        auto &context_data = *context_.get_context_data(encrypted.parms_id());
+        auto &parms = context_data.parms();
+        auto &coeff_modulus = parms.coeff_modulus();
+        size_t coeff_modulus_size = coeff_modulus.size();
+
+        auto norm = invariant_noise_internal(encrypted);
+
+        for (size_t i = 0; i < coeff_modulus_size; i++) {
+            auto power = static_cast<double>(sizeof(uint64_t) * 8 * i);
+            auto word = static_cast<double>(norm.get()[i]);
+            invariant_noise += word * exp2(power);
```

Comment:
Looking through other code, I suspect this is the wrong way to reduce a coefficient into a double precision number as this doesn't take into account the coefficients.

Reply:
Nevermind, I now think this is correct. The coefficients are indeed stored as RNS, which is what led to my doubt. However, `invariant_noise_internal` (the code I factored out of the existing `invariant_noise_budget` function) calls `compose_array(noise_poly, coeff_count, pool_)` before computing the infinity norm. This converts the RNS to multi-precision, so the infinity norm is a multi-precision value, which is what I was assuming here.